### ISSUE TITLE
doctest extension fails in comparing unicode output.

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#844]
The document listed at the end fails on `make doctest`, with the message:

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 188: ordinal not in range(128).

The module is available at http://www.sympy.org. The function "pprint()" produces unicode output. The output at the end of this report is slightly different from the output from terminal, since I had to format the former to look useful in the preview. The sympy module can be easily installed with pip and has no dependencies. So hopefully, this can be easily recreated.

Sphinx error report is attached as a file to this issue.

Using Python 2.7, Sphinx 1.1.2 and Sympy 0.7.1 on Ubuntu 11.10.

```
Document
---------------

.. testsetup:: *

  from __future__ import division
  from sympy import *

  x, y, z, t = symbols('x y z t')
  k, m, n = symbols('k m n', integer=True)
  f, g, h = symbols('f g h', cls=Function)
  init_printing(use_unicode=True)

.. doctest:: 

  >>> pprint(Integral(x, (x, 1, 2)))
  2     
  ⌠     
  ⎮ x dx
  ⌡     
  1 
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/844
- Originally reported by: [phn](https://bitbucket.org/phn)
- Originally created at: 2011-12-29T12:49:09.219

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-01-29 10:58:30+00:00_

Closes #844: Fix crashes when dealing with Unicode output in doctest extension.

→ <<cset 0c316f22eb63f1fb95185d0972a9e60f2f81caa3>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-01-29 10:59:22+00:00_

Note that the test still doesn't pass because sympy produces trailing whitespace, which docutils discards. You'll have to add the doctest.NORMALIZE_WHITESPACE option.
